test/integration/test_remote_files.py

```diff
@@ -115,6 +118,43 @@ def test_fetch_from_import(self):
 
         assert os.path.exists(os.path.join(self.library_dir, "a"))
 
+    def test_fetch_from_drs(self, celery_worker):
+        # This test needs an additional worker because the md5 hashing task must run in parallel
+        # with the data_fetch task. Otherwise, the data_fetch task runs first, and polls the DRS
+        # endpoint, which in turn dispatches the md5 task and returns a 204 Try Later. The data_fetch
+        # task dutifully tries again but the md5 task still hasn't run, so the drs endpoint dispatches
+        # another md5 ask and so on till the DRS fetch task eventually times out. Once the data_fetch
+        # task exits, all the md5 tasks run. Multiple workers prevent this from happening.
+        CONTENT = "a\n"
+        history_id = self.dataset_populator.new_history()
+
+        def create_drs_object():
+            hda = self.dataset_populator.new_dataset(history_id, content=CONTENT, wait=True)
```

Comment:
I think you can force a hash evaluation here. 
```
        self.dataset_populator.compute_hash(hda["id"])
```

I think it is also possible to upload with a hash (https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy_test/api/test_libraries.py#L316), but you might have to work a little harder to go down that route.

Reply:
Thanks for that! I thought this would be a lot harder but manually computing the hash turned out be trivial and it all looks ok now.